@patch('hive_service.check_fs_root')
@patch('time.time')
@patch('socket.socket')
def test_socket_timeout(self, socket_mock, time_mock, check_fs_root_mock):
    s = socket_mock.return_value
    s.connect = MagicMock()
    s.connect.side_effect = socket.error('')
    time_mock.return_value = 1000
    try:
        self.executeScript((self.COMMON_SERVICES_PACKAGE_DIR + '/scripts/hive_server.py'), classname='HiveServer', command='start', config_file='default.json', hdp_stack_version=self.STACK_VERSION, target=RMFTestCase.TARGET_COMMON_SERVICES)
        self.fail('Script failure due to socket error was expected')
    except:
        self.assert_configure_default()
